### ISSUE TITLE
Element ordering influences `--wrapAttributes` rule

### ISSUE THREAD

**Issue by kwridan** [nicklockwood/SwiftFormat#1454]
Spotted a small discrepancy with the `--wrapAttributes` rule for types, where based on the element ordering it may get ignored.

## Configuration

- Version: 0.51.9
- Rules:

```
--rules wrapAttributes
--funcattributes prev-line
--typeattributes prev-line
--varattributes same-line
```

## Example 1 (expected):

```swift
let myClass = MyClass()

@objc
final class MyClass: NSObject {}
```

`@objc` is on the previous line per the configuration when `myClass` is declared before the class.

## Example 2 (not expected)

```swift
@objc final class MyClass: NSObject {}
let myClass = MyClass()
```

Seems declaring `myClass` after the class breaks the rule.

---

Are there perhaps other rules we may have declared that are conflicting / causing this? 

Thanks 🙏

**Comment by nicklockwood** [comment]
@kwridan thanks for reporting this. It's a bug in the rule where basically the `class` declaration followed immediately by a `let` makes it think that it's a `class let` declaration, and so it applies the `--varattributes` option instead of `--typeattributes`. Should be an easy fix.

**Comment by nicklockwood** [comment]
@kwridan fixed in 0.51.10

**Comment by kwridan** [comment]
Thanks @nicklockwood
